hibernate-core/src/main/java/org/hibernate/type/descriptor/java/DurationJavaDescriptor.java

```diff
@@ -30,15 +44,22 @@ public String toString(Duration value) {
 		if ( value == null ) {
 			return null;
 		}
-		return String.valueOf( value.toNanos() );
+		return new StringBuilder(22)
+				.append( value.getSeconds() )
+				.append( value.getNano() )
+				.toString();
```

Comment:
To my understanding, the `nano` part is the fraction of second. Its value could be any value from [0, 999_999_999]. Suppose we specify the Duration as `new Duration(24 * 3600, 0)`, this part would be zero. It seems the encoding and decoding would not work as expected (assuming the nano part is always of length of 9).

If I were the author, I would introduce a delimiter (seems '.' is a natural choice) instead.

Reply:
Yes, thanks, of course you're right. I'll fix it.

I'm 99% sure this code is never ever called (it's not used for rendering literal durations), but I'll fix it anyway :-)

Reply:
Fixed, I believe.